torch/ao/quantization/quantize.py

```diff
@@ -89,10 +90,20 @@ def _observer_forward_hook(self, input, output):
     """
     return self.activation_post_process(output)
 
-def register_activation_post_process_hook(module):
+def _observer_forward_pre_hook(self, input):
+    r"""Forward pre hook that calls observer on the output
+    """
+    return self.activation_post_process(input[0])
+
+def register_activation_post_process_hook(module, pre_hook=False):
```

Comment:
should probably change the name if it can be a pre or post process hook

Reply:
I'm not sure what to call it. Any suggestions?

Reply:
I think `activation_post_process` is a confusing name here since it can now be activation_post_process and activation_pre_process now... we might consider making it activation_processor or something but it will be a bigger refactor that need to rename all `activation_post_process` in the code base..

For now I think it's OK to make it register_activation_hook or just leave it there..

Reply:
how about `register_activation_pre_or_post_process_hook`

Reply:
just want to clarify the confusion here, `activation_post_process` is the name we used to mean "observer or fake_quantize" in the beginning and `register_activation_post_process_hook` means "register a hook for `activation_post_process`" not "register a `post process` hook for activation". although it also makes sense to interpret the name that way..

Reply:
Hmm, then technically shouldn't it be `register_activation_post_process_pre_or_post_hook`? Since `activation_post_process` is an atomic unit

Reply:
@jerryzh168 what do you think the function name should be in that case?

Reply:
I think either just keep it this way and refactor all occurrences of `activation_post_process` in the future or just `register_activation_hook`

Reply:
Ok, how about we keep this one as is for now then and leave the refactoring for the future?